contracts/SP1_SpokePool.sol

```diff
@@ -0,0 +1,168 @@
+    ) SpokePool(_wrappedNativeTokenAddress, _depositQuoteTimeBuffer, _fillDeadlineBuffer) {
+        verifier = _verifier;
+        helios = _helios;
+        acrossCallProgramVKey = _acrossCallProgramVKey;
+        hubPool = _hubPool;
+    }
+
+    /**
+     * @notice This can be called by an EOA to relay call data stored on the HubPool on L1 into this contract.
+     * @dev Consider making this an onlyOwner function so that only a priveleged EOA can relay state. This EOA
+     * wouldn't be able to tamper with the _publicValues.
+     * @param _publicValues L1 contract state we want to relay into this contract. Contains a message that will
+     * be treated as calldata for a delegatecall into this contract.
+     * @param _proofBytes Proof bytes for the public values.
+     * @param _head Head number related to _publicValues.
+     */
+    function receiveL1State(
+        bytes calldata _publicValues,
+        bytes calldata _proofBytes,
+        uint256 _head
+    ) external validateInternalCalls {
+        // Verify proof and public values match:
+        ISP1Verifier(verifier).verifyProof(acrossCallProgramVKey, _publicValues, _proofBytes);
+        ContractPublicValues memory publicValues = abi.decode(_publicValues, (ContractPublicValues));
+
+        // Verify Helios light client is aware of the state root containing the public values:
+        bytes32 executionStateRoot = IHelios(helios).executionStateRoots(_head);
+        if (executionStateRoot != publicValues.stateRoot) {
+            revert StateRootMismatch();
+        }
+
+        // Validate state is intended to be sent to this contract:
+        (address _hubPool, address _target, bytes memory _message) = abi.decode(
+            publicValues.contractOutput,
+            (address, address, bytes)
+        );
+        if (_hubPool != hubPool) {
+            revert NotHubPool();
+        }
+        if (_target != address(this)) {
+            revert NotTarget();
+        }
+
+        // Store proof to prevent replay attacks. Proof and publicValues should have a 1 to 1 relationship
+        // so we can either store the hash of the proof or the public values here.
+        bytes32 dataHash = keccak256(_publicValues);
+        if (verifiedProofs[dataHash]) {
+            revert AlreadyReceived();
+        }
+        verifiedProofs[dataHash] = true;
+        emit VerifiedProof(dataHash, msg.sender);
+
+        // Execute the calldata:
+        /// @custom:oz-upgrades-unsafe-allow delegatecall
+        (bool success, ) = address(this).delegatecall(_message);
+        if (!success) {
+            revert DelegateCallFailed();
+        }
+    }
+
+    function initialize(
+        uint32 _initialDepositId,
+        address _crossDomainAdmin,
+        address _withdrawalRecipient
+    ) public initializer {
+        __SpokePool_init(_initialDepositId, _crossDomainAdmin, _withdrawalRecipient);
+    }
+
+    function _bridgeTokensToHubPool(uint256, address) internal override {
```

Comment:
Same question as below: maybe this should revert just to protect against this being called erroneously?

Probably matters less here, since the `SpokePool` doesn't do any accounting on balances.